### PR TITLE
Tests for CinemaHall model and controller

### PR DESCRIPTION
**Pull request by agusiol**
Installed rspec and created tests for CinemaHall model and controller(only GET method so far)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,3 +38,10 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+
+gem 'rspec', '~> 3.0'
```
Comment (wmaciejak) [inline Gemfile:43]: RSpec might be also in `development and test` group :)

File: spec/controllers/cinema_halls_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    describe 'GET #show' do
+        it "shows cinema_hall when valid id" do
+            cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
+            get :show, params: {id: cinema_hall.id}
+            expect(response.status).to eq(200)
+        end
+
+        it "responds with error when not valid id" do
+            expect{get :show, params: {id: "bad_id"}}.to raise_error(ActiveRecord::RecordNotFound)
+        end
```
Comment (wmaciejak) [inline spec/controllers/cinema_halls_controller_spec.rb:13]: Try to cover those two tests in `context` block. If you are using `... when ...` inside `it` block it means that you can easily create `context` block without big effort. Consider this:

```ruby
context "when valid ID" do
  it "shows cinema_hall" do # => consider if this description is valid, cause I did not see that you testing inside this block that cinema_hall are shown, from my perspective you testing if response status is 200 :)
    cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
    get :show, params: {id: cinema_hall.id}
    expect(response.status).to eq(200)
  end
end

context "when not valid ID" do
  it "responds with error" do
    expect{get :show, params: {id: "bad_id"}}.to raise_error(ActiveRecord::RecordNotFound)
  end
end
```

File: spec/models/cinema_hall_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe CinemaHall, type: :model do #:type => :model is it the same?
```
Comment (wmaciejak) [inline spec/models/cinema_hall_spec.rb:3]: > #:type => :model is it the same?

This is "old syntax" of Ruby hash. Currently we are using `key: :value`.

File: spec/models/cinema_hall_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    subject {
```
Comment (wmaciejak) [inline spec/models/cinema_hall_spec.rb:4]: Very good usage of subject! Nice! :)

File: spec/models/cinema_hall_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    it "is invalid without capacity" do
+        subject.capacity = nil
+        expect(subject).to_not be_valid
+    end
+
+    it "is invalid with capacity as string" do
+        subject.capacity ="ABC"
+        expect(subject).to_not be_valid
+    end
```
Comment (wmaciejak) [inline spec/models/cinema_hall_spec.rb:27]: Consider this structure:
```ruby
describe "validations" do
  context "when all params are passed" do
    it 'creates hall with valid number and capacity' do   
      expect(subject).to be_valid  
    end
  end

  describe "#capacity" do
    context "when there is no capacity" do
        it "is invalid" do
          subject.capacity = nil        
          expect(subject).to_not be_valid
        end
     end
     
    context "when capacity is a string" do
        it "is invalid" do
          subject.capacity = "ABC"        
          expect(subject).to_not be_valid
        end
     end
  end
end
```

File: spec/controllers/cinema_halls_controller_spec.rb (hunk 2 above)
Comment (agusiol) [reply spec/controllers/cinema_halls_controller_spec.rb:13]: Will this test check if the cinema_hall was shown?
    
```
it "shows cinema_hall" do 
          cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
          get :show, params: {id: cinema_hall.id}
          expect(response.status).to eq(200)
          expect(response.body).to include("Kyoto")
end
```

Comment (wmaciejak) [reply spec/controllers/cinema_halls_controller_spec.rb:13]: Yes, for sure! But consider that this test 
```
  expect(response.body).to include("Kyoto")
```
checks if response generally includes the word "Kyoto". Better idea would to check if there is exactly this record - something like:
```
expect(JSON.parse(response.body)).to eq( { id: cinema_hall.id, name: cinema_hall.name, capacity: cinema_hall.capacity })
```

Anyway - it looks fine, I'm just considering improvements.

**Pull request merged**